pkgs/misc/emulators/wine/default.nix

```diff
@@ -67,4 +69,13 @@ in if wineRelease == "staging" then
     wineUnstable = wine-build wineBuild "unstable";
   }
 else
-  wine-build wineBuild wineRelease
+  (if wineRelease == "wayland" then
+    callPackage ./wayland.nix {
+      wineWayland = wine-build wineBuild "wayland";
+      inherit pulseaudioSupport vulkanSupport vkd3dSupport;
+
+      pkgArches = lib.optionals (wineBuild == "wine32" || wineBuild == "wineWow") [ pkgsi686Linux ] ++ lib.optionals (wineBuild == "wine64" || wineBuild == "wineWow") [ pkgs ];
+    }
+    else
```

Comment:
This indirection here is indeed weird. I know you only followed what staging does, but I would probably just toggle a wayland feature instead and do all the work in `base.nix`. Then `packages.nix` will provide all this information for you.